### PR TITLE
Suggestion for Workbook Update Connection

### PR DESCRIPTION
**Pull request by cmtoomey**
Per #148, an option for updating workbook connections. Having to iterate over each connection individually is sub-optimal, but core REST API only allows one connection update at a time. 

This only accounts for connection string updates and not ports, username, or password changes. 

Use case is DB migration. 

Also - I'm not sure why the two HTML files were included. No edits were made after the fork.

### CODE REVIEW

File: tableauserverclient/server/request_factory.py [hunk 1]
```diff
@@ -313,6 +312,13 @@ def publish_req_chunked(self, workbook_item, connection_credentials=None):
         parts = {'request_payload': ('', xml_request, 'text/xml')}
         return _add_multipart(parts)
 
+class WorkbookConnection(object):
+    def update_req(self, connection_item):
+        xml_request = ET.Element('tsRequest')
+        connection_element = ET.SubElement(xml_request, 'connection')
+        connection_element.attrib['serverAddress'] = connection_item.server_address.lower()
```
Comment (t8y8) [inline tableauserverclient/server/request_factory.py]: May as well add the other attributes that are safe:
port
password

File: docs/docs/api-ref.md [hunk 2]
```diff
@@ -180,6 +180,14 @@ Modifies a workbook. The workbook item object must include the workbook ID and o
 Workbooks.update(wb_item_object)
 ```
 
+### Update workbook connection
+
+Updates a workbook connection string. The workbook connections must be populated before they strings can be updated. 
+
+```py
+Workbooks.update_conn(workbook, workbook.connections[i]
```
Comment (t8y8) [inline docs/docs/api-ref.md]: missing a closing `)`

File: docs/docs/populate-connections-views.md [hunk 3]
```diff
@@ -34,6 +34,14 @@ server.workbooks.populate_connections(workbook)
 print([connection.datasource_name for connection in workbook.connections])
 ```
 
+## Update connections for workbooks
+
+```py
+server.workbooks.populate_connections(workbook)
+workbook.connections[i].server_address = 'new_endpoint'
```
Comment (t8y8) [inline docs/docs/populate-connections-views.md]: This is technically correct, but I don't like the pattern (Not your fault, the way we try to keep the models and the upstream in sync are a little wonky)

Can you document it like this:
```py
server.workbooks.populate_connections(workbook)
conn_to_update = workbook.connections[0]  # Or whichever you want, but i is context-less and can't be copy/pasted
conn_to_update.server_address = 'abcdefg'
conn_to_update.port = 1337
server.workbooks.update_conn(workbook, conn_to_update)
```

/cc @RussTheAerialist

File: tableauserverclient/server/request_factory.py (hunk 1 above)
Comment (cmtoomey) [inline tableauserverclient/server/request_factory.py]: No particular reason - I was just editing in parts that I needed to change. Probably a good idea to add the other ones too.

File: docs/docs/populate-connections-views.md (hunk 3 above)
Comment (cmtoomey) [inline docs/docs/populate-connections-views.md]: That works - I'll make those changes.

**Comment by cmtoomey** [comment]
Changes made to reflect @t8y8 comments

File: docs/docs/populate-connections-views.md [hunk 4]
```diff
@@ -38,8 +38,13 @@ print([connection.datasource_name for connection in workbook.connections])
...
 ```py
 server.workbooks.populate_connections(workbook)
-workbook.connections[i].server_address = 'new_endpoint'
-server.workbooks.update_conn(workbook, workbook.connections[i])
+conn_to_update = workbook.connections[0]
+conn_to_update.server_address = 'new_address'
+conn_to_update.server_port = 1234
+conn_to_update.username = 'username'
+conn_to_update.password = 'password'
+conn_to_update.embed_password = TRUE/FALSE
```
Comment (t8y8) [inline docs/docs/populate-connections-views.md]: Oh, this should just be `True` or `False` in Python, the all caps could be a different variable

File: tableauserverclient/server/request_factory.py [hunk 5]
```diff
@@ -312,7 +313,8 @@ def publish_req_chunked(self, workbook_item, connection_credentials=None):
         parts = {'request_payload': ('', xml_request, 'text/xml')}
         return _add_multipart(parts)
 
-class WorkbookConnection(object):
+
+    class WorkbookConnection(object):
```
Comment (t8y8) [inline tableauserverclient/server/request_factory.py]: Oops, I don't think this should be indented

**Comment by cmtoomey** [comment]
Current fixes complete

**Comment by LGraber** [comment]
Looks good to me. At some point we might rethink how our "update" works. It is supposed to figure out what needs to be updated (based on anything that has changed) but it has gotten lax. Thus you need updateConn. If not, you might just call "update".

**Comment by LGraber** [comment]
Thanks!!

**Comment by t8y8** [comment]
I edited your commit messages and merged. Congrats on your first accepted PR, @cmtoomey !!

**Comment by cmtoomey** [comment]
Yay - thanks @t8y8...now just to wait until it ships

**Pull request merged**